native_cpp/src/clock.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+#include <chrono>
+#include "config.h"
+
+namespace benchmark {
+
+/**
+ * @brief Clock class
+ *
+ */
+class Clock {
+
+	using SystemClock = std::chrono::high_resolution_clock;
+	using Tick = std::chrono::high_resolution_clock::time_point;
+
+private:
+	Tick _last;
+
+public:
+	/**
+   * @brief Construct a new Clock object
+   *
+   */
+	Clock() noexcept : _last{SystemClock::now()} {}
+	/**
+   * @brief Reset clock time
+   *
+   */
+	void tic() noexcept { _last = SystemClock::now(); }
+	/**
```

Comment:
Nitpick - Looks like some lines use tabs for indentation and others use spaces. Consistency would be ideal.